README.md
# PicoIDE

Script for faster and easier development in *nix shell.

To use this script, you need to install compiler and text editor. (g++ and vim, for example).

### Installation 

* How do i install it?

To install PicoIDE, you need any c++ compiler to be installed. Google "c++ compiler installation <YOUR LINUX DISTRO>" for help.

After that, download archive, and extract it into any folder.

When the archive extracted, enter into folder with extracted files and compile "installer.cpp", and then execute compiled file.

Follow the intructions on screen. 

Finally, you have installed PicoIDE. Congrats! Now yo can remove installer.cpp and do_not_rename_me.cpp.

To run PicoIDE, just type ./picoide.

Have fun!

### Questions

* I want to use PicoIDE on Windows/DOS, can i do this?
  * No, you can't unless you optimize script for windows command shell (For example, to clear screen in linux, you need to type "clear", but in windows you need to type "cls")

* Can I use PicoIDE on my Slackware/FreeBSD/Debian/Gentoo/OSX/Pidora/etc?
  * You can use it everywhere, where the linux kernel is installed(even on Android, but for that you need to install compiler, text editor, and busybox)
  
* Where can I need PicoIDE?
  * As it uses CLI, you will have better performance on slower devices, such as Raspberry Pi Model A. Also, somebody prefer to use CLI instead of X11 IDEs.
  
* Can i use PicoIDE to compile Pascal/C/Java/Assembly/etc.
 * Yes, you can, but you need to modify some lines of code, which used to execute and compile programms.
  





CHANGELOG
CHANGELOG OF PICOIDE
--------------------------
0.0/0:

-Working build.
--------------------------
0.1/0:

- Added installer.
- Added About and license.
--------------------------
0.1.2/0:

-Added config support.
--------------------------
0.2/0:

- First release on Github.
- Added notifications.
--------------------------
0.2/143:

-Updated installer
==========================
-----END OF CHANGELOG-----
==========================


installer.cpp
#include <iostream>
#include <string>
#include <fstream>
#include <cstdlib>

using namespace std;
int main()
{
    int inn;
	float ver;
	string file;
	string space = " ";
	string editor;	
	string compiler;
	string player;
	string sound;
	ofstream com (".compile.pide");
	ofstream str (".structure.pide");
	ofstream confA (".compiler");
	ofstream confB (".editor");
	ofstream confC (".notify");
	
    ver = 0.2;
	system("clear");
	
	cout << "╞══ Installer for PicoIDE " << ver << " (C) Red_Pixel Software 2015" << endl;
	cout << "Welcome to PicoIDE installer! " << endl;
	cout << "Compile PicoIDE? (1-Yes/0-No): ";
    cin >> inn;
    if(inn==1){
        cout << "Type your C++ compiler or its location and press Enter: ";
        cin >> compiler;
        file = compiler + " do_not_rename_me.cpp -o picoide";
	    com << file;
	    com.close();
	    system("chmod +x .compile.pide");
	    system("./.compile.pide");
	    cout << " " << endl;
	    if(inn == 1){
	    confA << compiler + space;
	    confA.close();
	    cout << "Type your text editor or its location and press Enter: ";
	    cin >> editor;
	    cout << " " << endl;
 	    confB << editor + space;
	    confB.close();
	    cout << " Type your player or its location and press Enter " << endl;
	    cout << "If you want to disable this feature, type disable: ";
	    cin >> player;
	    if(player!="disable"){
	    cout << " " << endl;
	    cout << "Type your notificaton sound and press Enter: ";
	    cin >> sound;
	    cout << " " << endl;
	    confC << player + " -q " + sound;
	    confC.close();
	    system("chmod +x .notify");
	    }else if(player=="disabled"){
	    confC << " ";
	    confC.close();
	    system("chmod +x .notify");
	    }
	    if(ver==255){
	    cout << "Create or use internal C++ structure file?(1-Create/0-Internal)" << endl;
	    cin >> inn;
	    if(inn==1){
		system("rm .structure.pide");
	    system("nano .structure.pide");
	    }else{
	    }
	    system("chmod +x .notify");
		}else{
		}
	    cout << "That's all! Have a nice day <3" << endl;
	    }else ver++;
    }else{
        if(inn!=1){
        	cout << "Type your compiler or its location and press Enter: ";
        	cin >> compiler;
	        confA << compiler + space;
	        confA.close();
        }else cout << "Compiler already selected." << endl;
        cout << "Type your text editor or its location and press Enter: ";
        cin >> editor;
        cout << " " << endl;
        confB << editor + space;
	confB.close();
	cout << "Type your player or its location and press Enter: " << endl;
        cout << "If you want to disable this feature, type disable: ";
	    cin >> player;
	    if(player!="disable"){
	    cin >> sound;
	    cout << " " << endl;
	    confC << player + " -q " + sound;
	    confC.close();
	    system("chmod +x .notify");
	    }else if(player=="disabled"){
	    confC << " ";
	    confC.close();
	    system("chmod +x .notify");
	    }
	cout << "That's all! Have a nice day <3" << endl;
	    }	 
}


do_not_rename_me.cpp
#include <iostream>		//TODO:: Config from file - Done
#include <string>		//TODO:: Notifications - Done
#include <fstream>		//TODO:: Installer - Done
#include <cstdlib>		//TODO:: Autostructure +- 

using namespace std;
int main()
{
	float ver = 0.2;
	int build = 143;

//	V A R I A B L E S

	string editor;	
	string compiler;
	string file;
	string filee;
    string f;
	string wtf = "y";
	string ftw;
	int exit;
	fstream ftest;
	fstream str (".structure.pide");
	ofstream run (".executer.pide");
	ofstream com (".compile.pide");
	ifstream noty (".notify");
	ifstream comp (".compiler");
	ifstream edit (".editor");
	string project;

	comp >> compiler;
	edit >> editor;

	run << "clear" << endl;
	com << "clear" << endl;

	system("clear");
	cout << "╞══ Developed for easy development in linux console. See goo.gl/Czp9Hs for some  " << endl;
	cout << "╞══ details and documentation. Licensed under GNU GPL2. Uses some external tools." << endl;
	cout << "╞══        PicoIDE " << ver << "/" << build << " (C) Red_Pixel Software 2015     " << endl;
	cout << "╞════════════════════════════════════════════════════════════════════════════════" << endl;
	cout << "╞══ Enter your project name: ";
	cin >> project;
	cout << " " << endl;
	file = editor + " " + project;
	filee = compiler + " " + project + " -o built";
	run << file;
	com << filee;
	
	system("chmod +x .executer.pide");
	system("chmod +x .compile.pide");
	run.close();
	com.close();
	do{
	system("./.executer.pide");
	cout << "Compile?(Y/N) ";
	cin >> wtf;
	if(wtf=="Y"){
		system("./.compile.pide");
		system("./.notify");
	}else if(wtf=="y"){
		system("./.compile.pide");
		system("./.notify");
	}else{
		cout << "Exit from IDE?(Y/N) ";
		cin >> ftw;
		if(ftw=="y"){
		exit = 228;
		}else if(ftw=="Y"){
		exit = 228;
		}else if(ftw!="y"){
		exit = 0;
		}
	}
	cout << "" << endl;
	if(wtf=="y"){
	    cout << "Execute?(Y/N) ";
	    cin >> wtf;
	if(wtf=="Y"){
		system("./built");
		cin >> f;
	}else if(wtf=="y"){
		system("./built");
		cin >> f;
	    }else system("./.executer.pide");
	    }else if(wtf=="y"){
		    cout << "Execute?(Y/N) ";
	        cin >> wtf;
	if(wtf=="Y"){
		system("./picoide.co");
	}else if(wtf=="y"){
		system("./picoide.co");
	}else system("./.executer.pide");
	}
	}while(exit != 228);
	system("rm .executer.pide");
	

	return 0;
}

